webapi/src/main/scala/org/knora/webapi/messages/v1/responder/groupmessages/GroupMessagesV1.scala

```diff
@@ -32,29 +31,46 @@ import spray.json.{DefaultJsonProtocol, JsonFormat, NullOptions, RootJsonFormat}
 /**
   * Represents an API request payload that asks the Knora API server to create a new group.
   *
-  * @param name               the name of the group to be created (unique).
-  * @param description        the description of the group to be created.
-  * @param belongsToProject   the project inside which the group will be created.
-  * @param status             the status of the group to be created.
-  * @param hasSelfJoinEnabled the status of self-join of the group to be created.
+  * @param name        the name of the group to be created (unique).
+  * @param description the description of the group to be created.
+  * @param project     the project inside which the group will be created.
+  * @param status      the status of the group to be created (active = true, inactive = false).
+  * @param selfjoin    the status of self-join of the group to be created.
   */
 case class CreateGroupApiRequestV1(name: String,
                                    description: Option[String],
-                                   belongsToProject: IRI,
-                                   status: Boolean = true,
-                                   hasSelfJoinEnabled: Boolean = false) {
-    def toJsValue = GroupV1JsonProtocol.createGroupApiRequestV1Format.write(this)
+                                   project: IRI,
+                                   status: Boolean,
+                                   selfjoin: Boolean) extends GroupV1JsonProtocol {
+
+    def toJsValue = createGroupApiRequestV1Format.write(this)
 }
 
 /**
-  * Represents an API request payload that asks the Knora API server to update one property of an existing group.
+  * Represents an API request payload that asks the Knora API server to update an existing group.
   *
-  * @param propertyIri the property of the group to be updated.
-  * @param newValue    the new value for the property of the group to be updated.
+  * @param name        the new group's name.
+  * @param description the new group's description.
+  * @param status      the new group's status.
+  * @param selfjoin    the new group's self-join status.
   */
-case class UpdateGroupApiRequestV1(propertyIri: String,
-                                   newValue: String) {
-    def toJsValue = GroupV1JsonProtocol.updateGroupApiRequestV1Format.write(this)
+case class ChangeGroupApiRequestV1(name: Option[String] = None,
+                                   description: Option[String] = None,
+                                   status: Option[Boolean] = None,
+                                   selfjoin: Option[Boolean] = None) extends GroupV1JsonProtocol {
+
+    val parametersCount = List(
+        name,
+        description,
+        status,
+        selfjoin
+    ).flatten.size
+
```

Comment:
yes, but I opted for the number of parameters so that it is the same as in the other two case classes.